videostore/videostore.go

```diff
@@ -497,3 +518,79 @@ func (vs *videostore) Close() {
 		}
 	}
 }
+
+// GetStorageState returns the current storage state for this VideoStore.
+func (vs *videostore) GetStorageState() (*StorageState, error) {
+	storagePath := vs.config.Storage.StoragePath
+	usedBytes, err := getDirectorySize(storagePath)
+	if err != nil {
+		return nil, err
+	}
+	usedGB := float64(usedBytes) / float64(gigabyte)
+	limitGB := float64(vs.config.Storage.SizeGB)
+
+	fsUsage, err := diskusage.Statfs(storagePath)
+	if err != nil {
+		return nil, err
+	}
+	remainingGB := float64(fsUsage.AvailableBytes) / float64(gigabyte)
+
+	files, err := getSortedFiles(storagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		ranges                 []VideoTimeRange
+		currentFrom, currentTo time.Time
+		seenFirstSegment       bool
+	)
+
+	for _, file := range files {
+		info, err := getVideoInfo(file.name)
+		if err != nil {
+			vs.logger.Warnf("skipping unfinished or corrupt file %s: %v", file.name, err)
+			continue
+		}
+
+		start := file.startTime
+		end := start.Add(info.duration)
+
+		if !seenFirstSegment {
+			currentFrom, currentTo = start, end
+			seenFirstSegment = true
+			continue
+		}
+
+		if !start.After(currentTo) {
+			// overlaps or touches
+			if end.After(currentTo) {
+				currentTo = end
+			}
+		} else {
+			// there is a gap, so we flush the previous range
+			ranges = append(ranges, VideoTimeRange{
+				From: currentFrom.UTC().Format(TimeFormat) + "Z",
+				To:   currentTo.UTC().Format(TimeFormat) + "Z",
+			})
+			currentFrom, currentTo = start, end
+		}
+	}
+
+	if seenFirstSegment {
+		// append the final range
+		ranges = append(ranges, VideoTimeRange{
+			From: currentFrom.UTC().Format(TimeFormat) + "Z",
+			To:   currentTo.UTC().Format(TimeFormat) + "Z",
+		})
+	}
+
+	return &StorageState{
+		DiskUsage: DiskUsageState{
+			StorageUsedGB:            usedGB,
+			StorageLimitGB:           limitGB,
+			DeviceStorageRemainingGB: remainingGB,
```

Comment:
are you sure that this will return the amount of storage space on the disk partition the ~/.viam/video-store directory has available to it and not the sum total of storage space ALL partitions have available to them?